beacon-chain/monitor/service.go

```diff
@@ -59,6 +84,187 @@ type Service struct {
 	lastSyncedEpoch             types.Epoch
 }
 
+// NewService sets up a new validator monitor instance when given a list of
+// validator indices to track.
+func NewService(ctx context.Context, config *ValidatorMonitorConfig, tracked []types.ValidatorIndex) (*Service, error) {
+	ctx, cancel := context.WithCancel(ctx)
+	r := &Service{
+		config:                      config,
+		ctx:                         ctx,
+		cancel:                      cancel,
+		TrackedValidators:           make(map[types.ValidatorIndex]interface{}, len(tracked)),
+		latestPerformance:           make(map[types.ValidatorIndex]ValidatorLatestPerformance),
+		aggregatedPerformance:       make(map[types.ValidatorIndex]ValidatorAggregatedPerformance),
+		trackedSyncCommitteeIndices: make(map[types.ValidatorIndex][]types.CommitteeIndex),
+	}
+	for _, idx := range tracked {
+		r.TrackedValidators[idx] = nil
+	}
+	return r, nil
+}
+
+// Start waits until the beacon is synced and starts the monitoring system.
+func (s *Service) Start() {
+	tracked := make([]types.ValidatorIndex, len(s.TrackedValidators))
+	i := 0
+	for idx := range s.TrackedValidators {
+		tracked[i] = idx
+		i++
+	}
+	sort.Slice(tracked, func(i, j int) bool { return tracked[i] < tracked[j] })
+	log.WithFields(logrus.Fields{
+		"ValidatorIndices": tracked,
+	}).Info("Started service")
```

Comment:
Should we log this at the end?

Reply:
In my runtime tests it was reassuring getting the confirmation at launch that I was tracking the right set of validators. Vide the reply above about changing this message.